cfme/tests/v2v/test_v2v_cancel_migrations.py

```diff
@@ -4,97 +4,92 @@
 import fauxfactory
 import pytest
 
+from cfme import test_requirements
+from cfme.cloud.provider.openstack import OpenStackProvider
 from cfme.fixtures.provider import rhel7_minimal
 from cfme.infrastructure.provider.rhevm import RHEVMProvider
 from cfme.infrastructure.provider.virtualcenter import VMwareProvider
 from cfme.markers.env_markers.provider import ONE_PER_TYPE
 from cfme.markers.env_markers.provider import ONE_PER_VERSION
-from cfme.utils.appliance.implementations.ui import navigator
-from cfme.utils.log import logger
 from cfme.utils.wait import wait_for
 
 pytestmark = [
     pytest.mark.provider(
-        classes=[RHEVMProvider],
+        classes=[RHEVMProvider, OpenStackProvider],
         selector=ONE_PER_VERSION,
-        required_flags=['v2v'],
-        scope="module"
+        required_flags=["v2v"],
+        scope="module",
     ),
     pytest.mark.provider(
         classes=[VMwareProvider],
         selector=ONE_PER_TYPE,
-        fixture_name='source_provider',
-        required_flags=['v2v'],
-        scope="module"
-    )
+        required_flags=["v2v"],
+        fixture_name="source_provider",
+        scope="module",
+    ),
+    pytest.mark.usefixtures("v2v_provider_setup")
 ]
 
 
-@pytest.mark.parametrize('form_data_multiple_vm_obj_single_datastore', [['nfs', 'nfs',
+@test_requirements.v2v
+@pytest.mark.parametrize('mapping_data_multiple_vm_obj_single_datastore', [['nfs', 'nfs',
     [rhel7_minimal, rhel7_minimal]]],
     indirect=True)
-@pytest.mark.parametrize('cancel_migration_after_percent', [1, 10, 50, 80])
-def test_dual_vm_migration_cancel_migration(request, appliance, v2v_providers, host_creds,
-                                        conversion_tags,
-                                        form_data_multiple_vm_obj_single_datastore, soft_assert,
-                                        cancel_migration_after_percent):
+def test_dual_vm_migration_cancel_migration(request, appliance, soft_assert, provider,
+                                            mapping_data_multiple_vm_obj_single_datastore):
     """
     Polarion:
-        assignee: ytale
-        initialEstimate: 1/4h
+        assignee: sshveta
+        initialEstimate: 1/2h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
         casecomponent: V2V
+        testSteps:
+            1. Add source and target provider
+            2. Create infra map and migration plan
+            3. Start migration plan and cancel.
     """
-    # TODO: Improve this test to cover cancel operation at various stages in migration.
     # This test will make use of migration request details page to track status of migration
-    infrastructure_mapping_collection = appliance.collections.v2v_mappings
-    mapping = infrastructure_mapping_collection.create(
-        form_data_multiple_vm_obj_single_datastore.form_data)
+    cancel_migration_after_percent = 50
```

Comment:
**Optional**: Sometimes after 50%, migration works very rapidly depending upon load osp have so better to reduce it to 10/20/30%.

Tests looks good now, I was thinking to extend this test to check https://bugzilla.redhat.com/show_bug.cgi?id=1666799. We don't need to check much on RHV side though we should check if vmlist of RHV contains vm we are migrating but for OSP we need to check everything - instance + volume + port, I am working on that currently then.

Reply:
:+1: